### PR TITLE
feat: add JSON attribute filtering to existing method (AM-12)

### PR DESCRIPTION
**Pull request by rebeccabrennan**
Access Management Service - filter resource returning permissions and filtering out read permissions. Need to now filter resourceJson so that only data with read permissions will be shown.

### CODE REVIEW

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 1]
```diff
@@ -71,24 +71,30 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
...
             return null;
         }
 
-        if (READ.isGranted(explicitAccess.getPermissions())) {
-            Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
-            attributePermissions.put("/", Permissions.fromSumOf(explicitAccess.getPermissions()));
+        Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
+
+        explicitAccess.forEach(explicitAccessRecord -> {
+            if (READ.isGranted(explicitAccessRecord.getPermissions())) {
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: @yarpo do we still need this? Can we just list all permissions we found in DB without applying any additional logic to it?

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 2]
```diff
@@ -30,14 +32,27 @@ public void setupTest() {
         resourceId = UUID.randomUUID().toString();
     }
 
+    @Test
+    public void filterResource_whenAddTwoRecordsWithSameResourceIdAndUserIdButDifferentAttribute_ReturnsTwo() {
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: Please remove `filterResource_` from method name. Feel free to apply same to other methods in touched test class. The reason is that that class will only contain tests targeting `filterResource` service method.

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java [hunk 3]
```diff
@@ -28,6 +29,23 @@ public static ExplicitAccessRecord createRecord(String resourceId,
             .build();
     }
 
+    public static ExplicitAccessRecord createAttributeRecord(String resourceId,
```
Comment (geek-soft) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: There is lots of duplicated code. Extend existing `createRecord` method. Overload if needed (having new method name for same in this case is bad idea IMO)

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 2 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: 👍

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java (hunk 3 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/FilterResourceResponse.java [hunk 4]
```diff
@@ -12,6 +12,6 @@
 @Builder
 public class FilterResourceResponse {
     private String resourceId;
-    private JsonNode data;
+    private Map<String, JsonNode> data;
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/FilterResourceResponse.java]: That seems wrong. We want to return JsonNode but without properties that consumer should not see.

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java [hunk 5]
```diff
@@ -23,6 +23,7 @@
     public static final String OTHER_ACCESSOR_ID = "b";
     public static final Set<Permission> EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS =
         Stream.of(CREATE, READ, UPDATE).collect(toSet());
+    public static final String ATTRIBUTE = "/test";
```
Comment (geek-soft) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java]: I am not sure that should be here. Attributes are at core of access management. Hiding it here among less important constants might be bad idea. It might make tests less readable.

Comment (rebeccabrennan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java]: I removed it in recent code 😄

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/FilterResourceResponse.java (hunk 4 above)
Comment (rebeccabrennan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/FilterResourceResponse.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 6]
```diff
@@ -68,27 +71,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
      * @param userId       (accessorId)
      * @param resourceId   resource id
      * @param resourceJson json
-     * @return resourceJson or null
+     * @return filtered resourceJson based on READ permissions or null if no READ permissions on resource
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: Better but result is not just `resourceJson`. It is whole envelope with resource ID, filtered JSON and map of permissions. Please update.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 7]
```diff
@@ -68,27 +71,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
...
+        explicitAccess.forEach(explicitAccessRecord -> {
+            attributePermissions.put(explicitAccessRecord.getAttribute(),
+                Permissions.fromSumOf(explicitAccessRecord.getPermissions()));
+            if (READ.isGranted(explicitAccessRecord.getPermissions())) {
+                filteredResource.set(explicitAccessRecord.getAttribute().replaceFirst("/", ""),
+                    resourceJson.at(JsonPointer.valueOf(explicitAccessRecord.getAttribute())));
+            }
+        });
+
+        if (filteredResource.toString().equals("{}")) {
```
Comment (yarpo) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: why don't we do like this:

```filteredResource.fields().hasNext()```

or something similar?

Comment (yarpo) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: ```JsonNodeFactory.instance.objectNode().isEmpty()```

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 8]
```diff
@@ -68,27 +71,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
...
-            attributePermissions.put("/", Permissions.fromSumOf(explicitAccess.getPermissions()));
+        Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
+        ObjectNode filteredResource = JsonNodeFactory.instance.objectNode();
 
+        explicitAccess.forEach(explicitAccessRecord -> {
+            attributePermissions.put(explicitAccessRecord.getAttribute(),
+                Permissions.fromSumOf(explicitAccessRecord.getPermissions()));
+            if (READ.isGranted(explicitAccessRecord.getPermissions())) {
+                filteredResource.set(explicitAccessRecord.getAttribute().replaceFirst("/", ""),
+                    resourceJson.at(JsonPointer.valueOf(explicitAccessRecord.getAttribute())));
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: I find pointless doing `JsonPointer.valueOf` in case there is string constructor.  Please remove.

To be fair underneath same processes happen so it should not have any performance overhead.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 7 above)
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: I do not feel comfortable relying on `toString` that is not covered by any contract. In next minor version implementation can change to break test. No logic should be build around `#toString()` method. Please find better way of checking object is empty.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 9]
```diff
@@ -68,27 +71,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
...
+            }
+        });
+
+        if (filteredResource.toString().equals("{}")) {
+            return null;
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: That is wrong. If filtered object is empty but permissions exist in DB the following should be returned:

```
{
  resourceId: "#1",
  data: null,
  permissions: {
    "/name": [DELETE]
  }
}
```

See: https://tools.hmcts.net/confluence/display/AM/005.1+Use+scenarios

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/ExplicitAccessRecord.java [hunk 10]
```diff
@@ -15,7 +15,7 @@
 
     private final Set<Permission> explicitPermissions;
 
-    @Builder // All args constructor is needs for builder. @SuperBuilder cannot be used because IDE does not support it
+    @Builder // All args constructor is needed for builder. @SuperBuilder cannot be used because IDE does not support it
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/ExplicitAccessRecord.java]: Good spot. Have you checked this is the only place with spelling issue?

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 1 above)
Comment (yarpo) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: READ is _ONLY_ used to display data, not to filter permissions. All explicit permissions we store in our db for a resource should be displayed.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 7 above)
Comment (geek-soft) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: Spotted same issue ;)

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java [hunk 11]
```diff
@@ -28,6 +29,23 @@ public static ExplicitAccessRecord createRecord(String resourceId,
...
 
+    public static ExplicitAccessRecord createRecord(String resourceId,
+                                                    String accessorId,
+                                                    Set<Permission> explicitPermissions,
+                                                    String attribute) {
+        return ExplicitAccessRecord.builder()
+            .resourceId(resourceId)
+            .accessorId(accessorId)
+            .explicitPermissions(explicitPermissions)
+            .accessType(TestConstants.ACCESS_TYPE)
```
Comment (yarpo) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: `TestConstants.ACCESS_TYPE` I'd rename it to: `TestConstants.EXPLICIT_ACCESS_TYPE`

Maybe not in this PR, but in the future. Especially when we start using other access types.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 12]
```diff
@@ -68,27 +71,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
...
-        if (READ.isGranted(explicitAccess.getPermissions())) {
-            Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
-            attributePermissions.put("/", Permissions.fromSumOf(explicitAccess.getPermissions()));
+        Map<String, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
+        ObjectNode filteredResource = JsonNodeFactory.instance.objectNode();
 
+        explicitAccess.forEach(explicitAccessRecord -> {
+            attributePermissions.put(explicitAccessRecord.getAttribute(),
+                Permissions.fromSumOf(explicitAccessRecord.getPermissions()));
+            if (READ.isGranted(explicitAccessRecord.getPermissions())) {
```
Comment (geek-soft) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: Does it take visibility inheritance into consideration? Is for example possible to achieve following output for given permissions settings?

```
Permissions:

{
  "": [READ],
  "/name": [DELETE]
}

Output:

{
  resourceId: "#1",
  data: {
    "age": 18,
    "address": {
      "city": "London",
      "postcode": "SE1"
    }
  },
  permissions: {
    "": [READ],
    "/name": [DELETE]
  }
}
```

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java [hunk 13]
```diff
@@ -14,7 +14,8 @@ private TestDataFactory() {
 
     public static ExplicitAccessRecord createRecord(String resourceId,
                                                     String accessorId,
-                                                    Set<Permission> explicitPermissions) {
+                                                    Set<Permission> explicitPermissions
```
Comment (geek-soft) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: Why is that change needed? Please avoid pointless changes to ease review.

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java [hunk 14]
```diff
@@ -28,6 +29,23 @@ public static ExplicitAccessRecord createRecord(String resourceId,
             .build();
     }
 
+    public static ExplicitAccessRecord createRecord(String resourceId,
+                                                    String accessorId,
+                                                    Set<Permission> explicitPermissions,
```
Comment (geek-soft) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: Can you swap attribute and permissions. It feels more natural to me to have attribute first. Happy to talk if you disagree.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 15]
```diff
@@ -31,23 +36,40 @@ void setupTest() {
...
+            createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS, "/name"));
 
         FilterResourceResponse result = ams.filterResource(ACCESSOR_ID, resourceId, DATA);
 
+        assertThat(result).toString().contains("test");
+        assertThat(result).toString().contains("name");
+    }
+
+    @Test
+    void whenRowExistWithAccessorIdAndResourceId_ReturnPassedFilteredJsonObject() {
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: Wow. Having one test that covers filtering is way too little. It does me 0 (zero) confidence. Please largely expand on unit tests. Cover scenarios from confluence page and more.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 16]
```diff
@@ -31,23 +36,40 @@ void setupTest() {
...
-        ams.createResourceAccess(createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS));
+    void whenAddTwoRecordsWithSameResourceIdAndUserIdButDifferentAttribute_ReturnsTwoAttributes() {
+        ams.createResourceAccess(
+            createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS, "/test"));
+        ams.createResourceAccess(
+            createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS, "/name"));
 
         FilterResourceResponse result = ams.filterResource(ACCESSOR_ID, resourceId, DATA);
 
+        assertThat(result).toString().contains("test");
```
Comment (geek-soft) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: I beg you, do not use `toString` method in tests!!! I am not even sure what that tests is checking. A word `test` can appear in many places of `FilterResourceResponse`.

Looking for analogy if I was to have test that assures me that Becky is Becky I would not write it as `becky.has("green").has("long")` because these assertion could describe for example green eye or green scarf or something.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 6 above)
Comment (rebeccabrennan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/ExplicitAccessRecord.java (hunk 10 above)
Comment (rebeccabrennan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/ExplicitAccessRecord.java]: 👍

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java (hunk 13 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: 👍

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java (hunk 14 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: 👍

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java (hunk 11 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestDataFactory.java]: 👍

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java (hunk 16 above)
Comment (rebeccabrennan) [reply am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 7 above)
Comment (rebeccabrennan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 9 above)
Comment (rebeccabrennan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 6 above)
Comment (rebeccabrennan) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: 👍

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 12 above)
Comment (geek-soft) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: Since filtering is getting more complex having whole logic in the service does not make much sense. I suggest extracting JSON filtering logic to separate class.

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 17]
```diff
@@ -32,33 +33,37 @@ void setupTest() {
     }
 
     @Test
-    void whenRowExistWithAccessorIdAndResourceIdReturnPassedJsonObject() {
+    void whenRowExistsAndHaveReadPermissionsReturnEnvelopeWithData() {
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]:
```suggestion
    void whenRowExistsAndHasReadPermissionsShouldReturnEnvelopeWithData() {
```

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 18]
```diff
@@ -32,33 +33,37 @@ void setupTest() {
...
-            .permissions(createPermissionsForWholeDocument(READ_PERMISSION))
+            .permissions(ImmutableMap.of(JsonPointer.valueOf(""), READ_PERMISSION))
             .build());
     }
 
     @Test
-    void whenRowNotExistWithAccessorIdAndResourceIdReturnNull() {
-        String nonExistingUserId = "ijk";
-        String nonExistingResourceId = "lmn";
+    void whenRowExistsAndDoesNotHaveReadPermissionsReturnEnvelopeWithoutData() {
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]:
```suggestion
    void whenRowExistsAndDoesNotHaveReadPermissionsShouldReturnEnvelopeWithoutData() {
```

File: am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java [hunk 19]
```diff
@@ -32,33 +33,37 @@ void setupTest() {
...
+            .resourceId(resourceId)
+            .data(null)
+            .permissions(ImmutableMap.of(JsonPointer.valueOf(""), CREATE_PERMISSION))
+            .build());
     }
 
     @Test
-    void whenRowExistsAndDoesntHaveReadPermissionsReturnNull() {
-        ams.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, ACCESSOR_ID, CREATE_PERMISSION));
+    void whenNoRowExistsReturnNull() {
```
Comment (swalker125) [inline am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/FilterResourceIntegrationTest.java]:
```suggestion
    void whenNoRowExistsShouldReturnNull() {
```

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java [hunk 20]
```diff
@@ -27,7 +27,9 @@
         Stream.of(CREATE, READ, UPDATE).collect(toSet());
     public static final Set<Permission> CREATE_PERMISSION = Stream.of(CREATE).collect(toSet());
     public static final Set<Permission> READ_PERMISSION = Stream.of(READ).collect(toSet());
-    public static final JsonNode DATA = JsonNodeFactory.instance.objectNode();
+    public static final JsonNode DATA = JsonNodeFactory.instance.objectNode()
+        .put("name", "John")
```
Comment (swalker125) [inline am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java]: is this needed?

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java [hunk 21]
```diff
@@ -93,33 +94,36 @@ public void revokeResourceAccess(ExplicitAccessMetadata explicitAccessMetadata)
...
+        if (explicitAccess.isEmpty()) {
             return null;
         }
 
-        if (READ.isGranted(explicitAccess.getPermissions())) {
-            Map<JsonPointer, Set<Permission>> attributePermissions = new ConcurrentHashMap<>();
-            attributePermissions.put(JsonPointer.valueOf(""), Permissions.fromSumOf(explicitAccess.getPermissions()));
+        Map<JsonPointer, Set<Permission>> attributePermissions = explicitAccess.stream().collect(
+            Collectors.toMap(
+                explicitAccessRecord -> JsonPointer.valueOf(explicitAccessRecord.getAttribute()),
```
Comment (swalker125) [inline am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: could we write getters that will return JsonPointer and Permissions.fromSumOf in the ExplicitAccessRecord model?

**Comment by swalker125** [comment]
filter logic otherwise looks really good 👍

File: am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java (hunk 20 above)
Comment (geek-soft) [reply am-lib/src/test/java/uk/gov/hmcts/reform/amlib/helpers/TestConstants.java]: I would not say needed. It is to make sample data more real.

File: am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java (hunk 21 above)
Comment (geek-soft) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: Sounds like a good idea. Saw it in other place, make sense to make it consistent.

Comment (geek-soft) [reply am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java]: Here is that change: https://github.com/hmcts/am-lib/pull/68/commits/a2a1973bf4152bd1d1a11b8eb80ef67169753c23
